libsolidity/ast/Types.cpp

```diff
@@ -3090,6 +3090,14 @@ string FunctionType::canonicalName() const
 	return "function";
 }
 
+string FunctionType::humanReadableName() const
+{
+	switch (m_kind) {
+		case Kind::Error: return "error";
+		default: return Type::toString();
```

Comment:
```suggestion
        case Kind::Event: return "event " + m_declaration->name();
		default: return Type::toString();
```
We could add this extra case for Events as well as mentioned in https://github.com/ethereum/solidity/issues/12896#issuecomment-1095438214